### ISSUE TITLE
Tag themes with icon-theme or color-theme

### ISSUE THREAD

**Issue by waderyan** [Microsoft/vscode-vsce#121]
We want to be able to differentiate in our queries between icon-themes and color-themes only. Themes can be both, so they should be tagged with both.

**Comment by joaomoreno** [comment]
What are the exact tag names?

**Comment by waderyan** [comment]
`icon-theme` and `color-theme` seem good to me. 

We could possibly drop the `-theme` but `color` doesn't seem good.

**Comment by joaomoreno** [comment]
But all the current themes lack the `color-theme` tag. Are we going to go through them all in the Marketplace as before?

**Comment by waderyan** [comment]
That seemed to be @aeschli's suggestion. However, we don't to tag `color-theme`. We could simply add `icon` to the themes with icons and leave the rest of the themes as is. 

I believe the logic is simple. 

1. Is it a color only theme? Query by category:theme with no icon tag. 
2. Is it icons? Query for category:theme with icon tag

**Comment by joaomoreno** [comment]
Then the problem becomes how to query only the color themes? There doesn't seem to be a negation on the tags to be able to query on all themes without the tag `icon`.

**Comment by waderyan** [comment]
ping @hamenon see above. Is it possible with the api to query a negation on the tags?

**Comment by hamenon** [comment]
There is no negation option, but we do have an exact match option. So if you search for the following

- tag:"color theme" : It'll show you only extensions that have the exact tag color theme
- tag:"icon theme": It'll show only extensions that have the exact tag icon theme

I would suggest keeping things simple and just searching based on tag (and not mixing in the category) since it gives you the exact set of extensions you are looking for.

**Comment by joaomoreno** [comment]
The problem is that all the current themes in the marketplace don't have such a tag.

**Comment by hamenon** [comment]
We can possibly re-run the tag seeding job like we had earlier for the existing extensions. I don't believe that covered themes, @waderyan can correct me here if I'm wrong.

**Comment by waderyan** [comment]
@joaomoreno is there an indication in the manifest file that a particular extension is an icon theme? I looked on [vnext](https://vscode-vnext.azurewebsites.net/docs/extensionAPI/extension-manifest) but couldn't see anything in the docs. We will need something to indicate that the theme has icons. 

Once we know that we could run an algorithm on all existing extensions and tag them

1. Colorization only - `color-theme`
2. Icons only - `icon-theme`
3. Colorization and icons - `theme`

Then we could do all the queries. 

@hamenon what we need to do is run the tagging algorithm for all the existing themes and tag them as `color-theme`. 

Then we need to change vsce to tag new themes as `color-theme` or `icon-theme` or, in the case of an extension with both icons and color, tag as both `color-theme` and `icon-theme`.

**Comment by waderyan** [comment]
Ping @seanmcbreen see above if you'd like to weigh in.

**Comment by joaomoreno** [comment]
cc @aeschli 

AFAIK this is how we distinguish an icon theme from otherwise:

```
	"contributes": {
		"iconThemes": [
			{
				"id": "seti",
				"label": "Seti UI",
				"path": "./icons/seti-icon-theme.json"
			}
		]
	}
```

👍  on the proposal; I can make the changes in `vsce` today.

**Comment by joaomoreno** [comment]
Released in 1.9.0.
